go-apps/meep-platform-ctrl/server/user_authentication.go

```diff
@@ -54,6 +54,15 @@ func uaLoginUser(w http.ResponseWriter, r *http.Request) {
 	sessionStore := pfmCtrl.sessionMgr.GetSessionStore()
 	session, err := sessionStore.GetByName(username)
 	if err != nil {
+		// Check if max session count is reached before creating a new one
+		count := sessionStore.GetCount()
+		if count >= pfmCtrl.maxSessions {
+			err = errors.New("Maximum session count exceeded")
+			log.Error(err.Error())
+			http.Error(w, err.Error(), http.StatusInsufficientStorage)
```

Comment:
Error returned should be 503 Service Unavailable
https://www.restapitutorial.com/httpstatuscodes.html
503 Service Unavailable
The server is currently unable to handle the request **due to a temporary overloading** or maintenance of the server. The implication is that this is a temporary condition which will be alleviated after some delay. If known, the length of the delay MAY be indicated in a Retry-After header. If no Retry-After is given, the client SHOULD handle the response as it would for a 500 response.

Reply:
Updated as suggested.